lib/video_transcoder/videoTranscoding/transcoder.ex

```diff
@@ -0,0 +1,37 @@
+defmodule VideoTranscoderWeb.Transcoder do
+  use GenServer
+
+  # Client (methods)
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, [], opts)
+  end
+
+  # Server (callbacks)
+
+  @impl true
+  def init(init_arg) do
+    {:ok, init_arg}
+  end
+
+  @impl true
+  def handle_cast({:resolution, width, path, pid}, state) do
+    :timer.sleep(1500)
+    {:ok, path_to_transcoded_file} = transcode_and_save(path, width)
+    GenServer.cast(pid, {:created, path_to_transcoded_file})
+    {:noreply, state}
+  end
+
+  # Tools (additional methods)
+
+  def transcode_and_save(path, width) do
```

Comment:
This function should be private, right?
```suggestion
  defp transcode_and_save(path, width) do
```
Private functions have this benefit of throwing warnings if you remove the reference but forget to remove the function, so they tend to lead to a greater code quality in a longer run.